heka/sandbox/outputs/redshift.lua

```diff
@@ -125,20 +158,85 @@ local function bulk_load()
     end
 end
 
+function esc_str(v)
+    if v == nil then
+        return "NULL"
+    end
+    if string.len(v) > 65000 then
+        v = "TRUNCATED:" .. string.sub(v, 1, 65000)
+    end
+    return "'" .. con:escape(v) .. "'"
+end
+
+function esc_num(v)
+    if v == nil then
+        return "NULL"
+    end
+    if type(v) ~= "number" then
+        return esc_str(v)
+    end
+    return tostring(v)
+end
+
+function esc_ts(v)
+    if v == nil then
+        return "NULL"
+    end
+    if type(v) ~= "number" then
+        return esc_str(v)
+    end
+    local seconds = v / 1e9
+    return "(TIMESTAMP 'epoch' + " .. seconds .. " * INTERVAL '1 seconds')"
+end
+
 -- plugin interfaces
 function process_message()
     if sep == " " then
-        fh:write("INSERT INTO logs VALUES")
+        fh:write("INSERT INTO telemetry_sample_42 VALUES")
     end
-    fh:write(string.format("%s(%d,%d,%d,'%s','%s','%s','%s','%s')", sep,
-    read_message("Timestamp"),
-    read_message("Severity") or 7,
-    read_message("Pid") or 0,
-    con:escape(read_message("Hostname")),
-    con:escape(read_message("Type")),
-    con:escape(read_message("Logger")),
-    con:escape(read_message("EnvVersion")),
-    con:escape(read_message("Payload"))))
+    fh:write(string.format("%s(%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s)", sep,
+        esc_ts(read_message("Timestamp")),
```

Comment:
You will probably be a lot better off reading the entire raw message down and accessing it from the Lua table.